contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,244 @@
+pragma solidity ^0.4.24;
+
+import "./ITransferManager.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+/**
+ * @title Transfer Manager module for manually approving or blocking transactions between accounts
+ */
+contract BlacklistTransferManager is ITransferManager {
+    using SafeMath for uint256;
+
+    bytes32 public constant ADMIN = "ADMIN";
+    
+    struct BlacklistsDetails {
+        uint256 startDate;
+        uint256 endDate;
+        uint256 repeatPeriodInDays;
+    }
+
+    //hold the different blacklist corresponds to name of blacklist type
+    mapping(bytes32 => BlacklistsDetails) blacklists;
+
+    //hold the blacklisted address corresponds to the blacklist type
+    mapping(address => bytes32[]) investorToBlacklist;
+
+    //get list of the addresses for a particular blacklist
+    mapping(bytes32 => address[]) blacklistToAddress;
+
+    // Emit when new blacklist type is added
+    event AddBlacklistType(
+        uint256 _startDate, 
+        uint256 _endDate, 
+        bytes32 _name, 
+        uint256 _repeatPeriodInDays
+    );
+    
+    // Emit when there is change in the blacklist type
+    event ModifyBlacklistType(
+        uint256 _startDate,
+        uint256 _endDate, 
+        bytes32 _name, 
+        uint256 _repeatPeriodInDays
+    );
+    
+    // Emit when the added blacklist type is deleted
+    event DeleteBlacklistType(
+        bytes32 _name
+    );
+
+    // Emit when new investor is added to the blacklist type
+    event AddInvestorToBlacklist(
+        address _investor, 
+        bytes32 _blacklistName
+    );
+    
+    // Emit when investor is deleted from the blacklist type
+    event DeleteInvestorFromBlacklist(
+        address _investor,
+        bytes32 _blacklist
+    );
+
+    /**
+     * @notice Constructor
+     * @param _securityToken Address of the security token
+     * @param _polyAddress Address of the polytoken
+     */
+    constructor (address _securityToken, address _polyAddress)
+    public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /**
+    * @notice This function returns the signature of configure function
+    */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+
+    /// @notice Used to verify the transfer transaction 
+    function verifyTransfer(address _from, address /* _to */, uint256 /* _amount */, bool /* _isTransfer */) public returns(Result) {
+        if(!paused){
+            if(investorToBlacklist[_from].length != 0){
+                for(uint256 i = 0; i < investorToBlacklist[_from].length; i++){
+                    uint256 blacklistDate = ((blacklists[investorToBlacklist[_from][i]].endDate)
+                    .sub(blacklists[investorToBlacklist[_from][i]].startDate))
+                    .add(blacklists[investorToBlacklist[_from][i]].repeatPeriodInDays * 1 days);
+                    uint256 repeater = now.div((blacklists[investorToBlacklist[_from][i]].endDate)
```

Comment:
Could you add a comment here explaining the algorithm a bit more. If we wanted to modify the repeat to be from the startTime rather than the endTime would that be hard?